routes/user.js

```diff
@@ -0,0 +1,187 @@
+const router = require("express").Router();
+const bcrypt = require("bcryptjs");
+const Error = require("../models/Error");
+const Joi = require("@hapi/joi");
+const jwtService = require("../services/JwtService");
+
+const userService = require("../services/UserService");
+
+router
+    .route("/")
+    /*
+     * GET USER INFO
+     * [Only loggedIn user can see their info. The userID of loggedIn user is stored in the JWT token itself]
+     */
+    .get(async (req, res) => {
+        try {
+            const { userID } = jwtService.verify(req.headers["authorization"]);
+            if (userID == undefined) {
+                res.status(403).send(new Error("FORBIDDEN", "Unauthorized"));
+            } else {
+                const user = await userService.getUser(userID);
+                res.status(200).json({
+                    code: "OK",
+                    result: "SUCCESS",
+                    user: {
+                        zprn: user.zprn,
+                        firstName: user.firstName,
+                        lastName: user.lastName,
+                        department: user.department,
+                    },
+                });
+            }
+        } catch (error) {
+            res.status(500).send(new Error("INTERNAL_SERVER_ERROR", error));
+        }
+    })
+    /*
+     * ADD NEW USER
+     */
+    .post(async (req, res) => {
```

Comment:
why aren't we having a `api/users/add` or maybe `api/users/create` endpoint for this? It will be much more clearer that way?

Reply:
Okay